doc/manual/src/glossary.md

```diff
@@ -1,11 +1,95 @@
 # Glossary
 
   - [derivation]{#gloss-derivation}\
-    A description of a build task. The result of a derivation is a
-    store object. Derivations are typically specified in Nix expressions
-    using the [`derivation` primitive](language/derivations.md). These are
-    translated into low-level *store derivations* (implicitly by
-    `nix-env` and `nix-build`, or explicitly by `nix-instantiate`).
+    A [build] task declared in the [Nix language](./language/index.md) using the built-in function [`derivation`](language/derivations.md).
+
+    [derivation]: #gloss-derivation
+
+  - [build]{#gloss-build}\
+    Produce a [store object].
+
+    See [`nix-build`](./command-ref/nix-build.md).
+
+    See [`nix build`](./command-ref/new-cli/nix3-build.md) (experimental).
+
+    Nix uses a two-stage build process:
+    A Nix expression that evaluates to a [derivation] is [instantiate]d to a [store derivation], which is then [realise]d to a [store object].
+
+    ```
+    +--------------------+
+    | Nix language       |
+    | .----------------. |
+    | | Nix expression | |
+    | '----------------' |
+    |         |          |
+    |      evaluate      |
```

Comment:
I find this explanation and diagram slightly confusing because "evaluate" is not a definition in this document, unlike `instantiate` and `realise`: it looks like it is yet another command and stage.

I would have expected "Nix expression of type `derivation`" rather than "Nix expression - evaluate -> derivation".

Reply:
There is no such thing as a type `derivation`, and even the Nix language concept of a derivation is vague. It's an attribute set with two magic attributes `drvPath` and `outPath`, where the latter is also used as the value in string interpolation. See https://github.com/NixOS/nix/pull/7332#discussion_r1029343720

I can add a definition for evaluation though, but maybe in a different PR to keep this one small.